src/osd/PGLog.h

```diff
@@ -1266,8 +1267,10 @@ struct PGLog : DoutPrefixProvider {
     for (auto p = entries.begin(); p != entries.end(); ++p) {
       invalidate_stats = invalidate_stats || !p->is_error();
       if (log) {
-	ldpp_dout(dpp, 20) << "update missing, append " << *p << dendl;
-	log->add(*p);
+       log->add(*p);
+       ldpp_dout(dpp, 20) << "update missing, appended " << *p << dendl;
+       if (should_rollforward)
```

Comment:
This seems like it should be outside of the if(log) check.  Also, add braces.

Reply:
@athanatos but we specifically want that check this itself, since we only want to rollforward, if actually `log->add` operation happened.